ecosystem/sep-0009.md

```diff
@@ -85,6 +85,7 @@ Name | Type          | [Format](#encodings) |Description
 `sex` | string        | | `male`, `female`, or `other`                                                                
 `proof_of_income` | binary        | | Image of user's proof of income document
 `proof_of_liveness` | binary	| | video or image file of user as a liveness proof
+`account_id` | string | [Account Identifier Format](#account-identifier-format) | An idenfier for the user's off-chain account. Could be a bank, virtual, or other account type.
```

Comment:
I fear we may need to keep adding new fields like this. To prevent that we could make this field an array of strings, so the anchor can request any number of fields and if something fails, they can explain what's wrong in the response body.

Also, `account` is too broad as it can be mis-understood as the user Stellar Account. Here's my suggestion:

```suggestion
`offchain_account_ids` | string[] | [Account Identifier Format](#account-identifier-format) | An array of idenfiers for the user's off-chain account. Could contain one or more identifiers of the type bank, virtual, or other(s).
```

Reply:
Thanks Marcelo, adding an `offchain` prefix is fine by me.

I share your concern about needing multiple `offchain_account_id` fields for different accounts. This would be the first time we add an array type to the standard though, and it raises questions about how multiple values are encoded in different content types.

JSON is straight-forward:

```js
{
    "offchain_account_ids": ["cvu:X", "cbu:Y"]
}
```

And most web frameworks handle the two variations that exist for URL parameters:

```
?a=1&a=2&a=3
?a=1,2,3
```

But I don't think the same is true for `application/x-www-form-urlencoded`, which SEP-12 requires support for. We could address this in a later PR though.

Reply:
After further thought, I believe it's easier for wallets and for users if we go with a `string` type with comma-separated values. It should make it way easier for wallets to implement that when constructing the form automatically, and also easier for users to fill the form directly, without the wallet having to process the content prior to submitting it to the Anchors.

Here is how it'd look like in [SEP-12 `GET /customer` response](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0012.md#response):

```json
{
   "id": "d1ce2f48-3ff1-495d-9240-7a50d806cfed",
   "status": "NEEDS_INFO",
   "fields": {
      "offchain_account_ids": {
         "description": "The identifiers needed in order to deliver funds to a person's bank account in Argentina. Should be a list of comma-separated values in the format 'cbu:X,cvu:Y', containing the Clave Bancaria Uniforme and Clave Virtual Uniforme, respectively.",
         "type": "string"
      }
   },
   "provided_fields": {
      "last_name": {
         "description": "The customer's last name",
         "type": "string",
         "status": "ACCEPTED"
      }
   }
}
```

Reply:
Hey @marcelosalloum I've been thinking about this a bit, and unfortunately this is blocking a partner so we need to come to a decision ideally today.

 > It should make it way easier for wallets to implement that when constructing the form automatically, and also easier for users to fill the form directly, without the wallet having to process the content prior to submitting it to the Anchors.

I doubt wallets will ask users to enter their account IDs as a comma separated list, since its a rather awkward UX, especially if they'd need to enter the schema prefixes themselves (like `cvu:123`). They'll either ask for one value of a particular type (either CVU or CBU) or provide some kind of dynamic UI where a user can add additional values via some (+) button or something similar.

So I don't think using a comma-separated list will simplify wallet processing logic in practice, although I do see what you were going for.

---
 
I also think asking for an array of values has its own issues. The first is that if we add an `array` value to SEP-12/9, how do we specify the type for the array's values? We'd need something like this:

```js
{
    "fields": {
        "offchain_account_id": {
            "description": "Account identifiers. Supported Identifiers include CVU & CBU numbers.",
            "type": "array",
            "items": {
                 "type": "string"
            }
        }
     }
}
```

Thats not a huge deal, but something potentially more impactful is that wallets won't know how to format & validate their users' input for these types of fields unless they know which formats a specific anchor supports ahead of time. If we used something like `cvu_number`, even if the wallet didn't have a relationship with the anchor, it would know how to format / validate the value.

---

So, I'm afraid that we're making a change that adds more complexity than it removes. If you agree, I'll move forward by scratching this proposal and adding `cvu_number` and `cbu_number` SEP-9 fields that can be used in SEP-12 just like all other existing fields. I think we will eventually need to figure out how to support multiple values for the same field, but I no longer believe using generic field names is advantageous.

Reply:
SGTM!